apps/hash-graph/lib/graph/src/api/rest/property_type.rs

```diff
@@ -96,12 +119,35 @@ async fn create_property_type<P: StorePool + Send>(
     pool: Extension<Arc<P>>,
     domain_validator: Extension<DomainValidator>,
     body: Json<CreatePropertyTypeRequest>,
-) -> Result<Json<ListOrValue<OntologyElementMetadata>>, StatusCode> {
-    let Json(CreatePropertyTypeRequest {
+) -> Result<Json<ListOrValue<OntologyElementMetadata>>, StatusCode>
+where
+    for<'pool> P::Store<'pool>: RestApiStore,
+{
+    let mut store = pool.acquire().await.map_err(|report| {
+        tracing::error!(error=?report, "Could not acquire store");
+        StatusCode::INTERNAL_SERVER_ERROR
+    })?;
+
+    #[allow(clippy::infallible_destructuring_match)]
+    let CreateOwnedPropertyTypeRequest {
         schema,
         owned_by_id,
         actor_id,
-    }) = body;
+    } = match body.0 {
+        CreatePropertyTypeRequest::Owned(request) => request,
+        #[cfg(feature = "type-fetcher")]
+        CreatePropertyTypeRequest::External(request) => {
+            return Ok(Json(ListOrValue::Value(
+                store
+                    .load_external_type(
+                        &domain_validator,
+                        OntologyTypeReference::DataTypeReference((&request.schema).into()),
```

Comment:
```suggestion
                        OntologyTypeReference::PropertyTypeReference((&request.schema).into()),
```
Similar to above, or is this intentional?